src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets

```diff
@@ -50,6 +50,34 @@
     </PropertyGroup>
   </Target>
 
+  <!--
+      Target to disable analyzers via an MSBuild property 'RunAnalyzers'. 
+      Note that we remove analyzers only for non-design time builds.
+      Design time builds are background builds inside Visual Studio,
+      see details here: https://github.com/dotnet/project-system/blob/master/docs/design-time-builds.md
+      We do not remove analyzers for design time builds to allow users to view the installed analyzers
+      in 'Analyzers' node in the solution explorer and also execute analyzers on-demand with "Run Code Analysis" command.
+      Instead, for design time builds, the VS IDE will read this MSBuild property and skip analyzer execution if it is 'false'.
+  -->
+  <Target Name="_DisableAnalyzers"
+          BeforeTargets="CoreCompile"
+          Condition="'$(RunAnalyzers)' != 'true'">
```

Comment:
Is RunAnalyzers always explicitly set somewhere else? I sort of figured it would default to 'true'